terraform/production/service_account_roles.tf

```diff
@@ -0,0 +1,200 @@
+  role = aws_iam_role.notary.id
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      # Reputational Encryption Key
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "kms:GetPublicKey",
+          "kms:Decrypt",
+          "kms:DescribeKey"
+        ],
+        "Resource" : "arn:aws:kms:us-east-1:588562868276:key/bc739d17-8569-4741-9385-9264715b90b6"
+      },
+      # Test Key
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "kms:GetPublicKey",
+          "kms:Decrypt",
+          "kms:DescribeKey"
+        ],
+        "Resource" : "arn:aws:kms:us-east-1:588562868276:key/e811e4c5-ea63-4da3-87d4-664dc5395169"
+      },
+      # S3 Full Access
+      {
+        "Effect" : "Allow",
+        "Action" : [
+          "s3:*",
+          "s3-object-lambda:*"
+        ],
+        "Resource" : "*"
+      }
+    ]
+  })
+}
+
+# The ServiceAccount to be used by the signing runner
+resource "kubectl_manifest" "notary_service_account" {
+  yaml_body = <<-YAML
+    apiVersion: v1
+    kind: ServiceAccount
+    metadata:
+      name: notary
+      namespace: pipeline
+      annotations:
+        eks.amazonaws.com/role-arn: ${aws_iam_role.notary.arn}
+  YAML
+  depends_on = [
+    aws_iam_role_policy.notary
+  ]
+}
+
+# The IAM role granting Spackbot full access to spack-binaries-prs S3 bucket.
+resource "aws_iam_role" "full_crud_access_spack_binaries_prs" {
+  name        = "FullCRUDAccessToBucketSpackBinariesPRs"
+  description = "Managed by Terraform. Grants Kubernetes pods access to read/write/delete objects from the spack-binaries-prs S3 bucket."
+  assume_role_policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Principal" : {
+          "Federated" : module.production_cluster.oidc_provider_arn
+        },
+        "Action" : "sts:AssumeRoleWithWebIdentity",
+        "Condition" : {
+          "StringEquals" : {
+            "${module.production_cluster.oidc_provider}:aud" : "sts.amazonaws.com"
+          }
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "put_spack_binaries_prs" {
+  name = "PutObjectsInBucketSpackBinariesPRs"
+  role = aws_iam_role.full_crud_access_spack_binaries_prs.id
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:PutObject",
+        "Resource" : "arn:aws:s3:::spack-binaries-prs/*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy" "delete_spack_binaries_prs" {
+  name = "DeleteObjectsFromBucketSpackBinariesPRs"
+  role = aws_iam_role.full_crud_access_spack_binaries_prs.id
+  policy = jsonencode({
+    "Version" : "2012-10-17",
+    "Statement" : [
+      {
+        "Effect" : "Allow",
+        "Action" : "s3:DeleteObject",
+        "Resource" : "arn:aws:s3:::spack-binaries-prs/*"
+      }
+    ]
+  })
+}
+
+resource "kubectl_manifest" "spackbot_service_account" {
```

Comment:
I think having the full role names in a comment here would be good just for searching through the code (as I tried that and failed as part of reviewing this PR), but otherwise LGTM

```suggestion

# Create the spackbot-spack-io and spackbotdev-spack-io ServiceAccounts
resource "kubectl_manifest" "spackbot_service_account" {
```